.github/workflows/check-invariant-test.yml

```diff
@@ -0,0 +1,40 @@
+name: Strict NoThunk test
+on:
+  push:
+    branches:
+      - master
+      - yogeshs/3001-check-tvar-invariant
+
+
+jobs:
+  build-test:
+    name: "Build and Test with Strict Check TVar Invariant Check"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.1
+        with:
+          repository: input-output-hk/ouroboros-network
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+          fetch-depth: 0
+
+      - name: Prepare nix
+        uses: cachix/install-nix-action@v17
+        with:
+          extra_nix_config: |
+            trusted-public-keys = iohk.cachix.org-1:DpRUyj7h7V830dp/i6Nti+NEO2/nhblbov/8MW7Rqoo= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+            substituters = https://cache.nixos.org https://hydra.iohk.io https://iohk.cachix.org
+      - name: Github cache ~/.cabal/packages, ~/.cabal/store and dist-newstyle
+        uses: actions/cache@v2.1.5
+        with:
+          path: |
+            ~/.cabal/packages
+            ~/.cabal/store
+            dist-newstyle
+          key: |
+            cabal-${{ runner.os }}-${{ hashFiles('cabal.project', 'default.nix', 'shell.nix') }}
+          restore-keys: |
+            cabal-${{ runner.os }}-${{ hashFiles('cabal.project', 'default.nix', 'shell.nix') }}
+
+      - name: "Build and Test"
+        run: |
+          nix-build -A tvar-invariant-checks.tests
```

Comment:
WDYT about adding `--keep-going` to ensure that we still run unrelated test suites to completion even if a previous one already failed?